lib/utils/aws/internal/region/codegen.go

```diff
@@ -0,0 +1,200 @@
+		} `json:"regions"`
+	} `json:"partitions"`
+}
+
+func main() {
+	partitionsContents, err := loadSDKPartitionsFile()
+	if err != nil {
+		log.Fatalf("failed to parse patitions: %s", trace.DebugReport(err))
+	}
+
+	data, err := convertToTemplateData(partitionsContents)
+	if err != nil {
+		log.Fatalf("failed to parse partitions file: %s", trace.DebugReport(err))
+	}
+
+	file, err := os.Create(outputFilename)
+	if err != nil {
+		log.Fatalf("failed to create output file: %v", err)
+	}
+	defer func() {
+		if err := file.Close(); err != nil {
+			log.Fatalf("failed to close output file: %v", err)
+		}
+	}()
+
+	if err := tmpl.Execute(file, data); err != nil {
+		log.Fatalf("failed to generate file: %v", err)
+	}
+}
+
+func convertToTemplateData(partitionsContents []byte) (*TemplateData, error) {
+	var awsPartitions AWSPartitions
+	if err := json.Unmarshal(partitionsContents, &awsPartitions); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if len(awsPartitions.Partitions) == 0 {
+		return nil, trace.BadParameter("generated an empty regions list. check if the partitions JSON file format has changed")
+	}
+
+	data := &TemplateData{
+		Regions:       make(map[string]string),
+		GlobalRegions: make(map[string]string),
+	}
+
+	for _, partition := range awsPartitions.Partitions {
+		for name, region := range partition.Regions {
+			if matchGlobalRegion.MatchString(name) {
+				data.GlobalRegions[name] = region.Description
+				continue
+			}
+
+			data.Regions[name] = region.Description
+		}
+	}
+
+	return data, nil
+}
+
+// loadSDKPartitionsFile loads the partitions JSON file from AWS SDK.
+func loadSDKPartitionsFile() ([]byte, error) {
+	cmd := exec.Command("go", "list", "-m", "-f", "{{.Dir}}", awsPackageName)
+	awsPackageDir, err := cmd.Output()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	contents, err := os.ReadFile(filepath.Join(strings.TrimSpace(string(awsPackageDir)), awsPartitionsFilePath))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return contents, nil
+}
+
+const (
+	// outputFilename is the generated filename.
+	outputFilename = "regions.go"
+	// awsPackageName is the AWS package used to find the paritions references.
+	awsPackageName = "github.com/aws/aws-sdk-go-v2"
+	// awsPartitionsFilePath is the patitions JSON file path inside the package.
+	awsPartitionsFilePath = "internal/endpoints/awsrulesfn/partitions.json"
+)
+
+var (
+	// matchGlobalRegion is a regex that defines the format of AWS global regions.
+	// Those regions are usually used for endpoint resolution.
+	//
+	// The regex matches the following from left to right:
+	// - `aws` prefix.
+	// - optional -us-gov, -cn, -iso, -iso-b for corresponding partitions
+	// - `global` suffix.
+	//
+	// Reference:
+	// https://github.com/aws/aws-sdk-go-v2/blob/main/codegen/smithy-aws-go-codegen/src/main/resources/software/amazon/smithy/aws/go/codegen/endpoints.json
+	matchGlobalRegion = regexp.MustCompile(`^aws(-us-gov|-cn|-iso|-iso-b|-iso-e|-iso-f)?-global$`)
```

Comment:
i think the regex is an overkill and will need to be maintained when new partition is added. `-global` suffix check is good enough IMO. non-global regions always end with digits.